gax/src/main/java/com/google/api/gax/batching/BatchEntry.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are
+ * met:
+ *
+ *     * Redistributions of source code must retain the above copyright
+ * notice, this list of conditions and the following disclaimer.
+ *     * Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following disclaimer
+ * in the documentation and/or other materials provided with the
+ * distribution.
+ *     * Neither the name of Google LLC nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.google.api.gax.batching;
+
+import com.google.api.core.BetaApi;
+import com.google.api.core.InternalExtensionOnly;
+import com.google.api.core.SettableApiFuture;
+import com.google.auto.value.AutoValue;
+import javax.annotation.Nullable;
+
+/**
+ * This class contains the element and it's corresponding unresolved future, which would be resolved
+ * when batch is {@link BatchingDescriptor#splitResponse successful} or {@link
+ * BatchingDescriptor#splitException failed}.
+ *
+ * @param <ElementT> The type of each individual element to be batched.
+ * @param <ElementResultT> The type of the result for each individual element.
+ */
+@BetaApi("The surface for batching is not stable yet and may change in the future.")
+@InternalExtensionOnly("For google-cloud-java client use only.")
+@AutoValue
+public abstract class BatchEntry<ElementT, ElementResultT> {
+
+  /** Get a new builder. */
+  public static <ElementT, ElementResultT> Builder<ElementT, ElementResultT> newBuilder() {
+    return new AutoValue_BatchEntry.Builder<>();
+  }
+
+  @Nullable
```

Comment:
When can this be null?

Reply:
As of now, we do not have any sanity checks on [BatcherImpl.add()](https://github.com/googleapis/gax-java/blob/f18673a1efb985243eee4a5cce01c7054ce18c23/gax/src/main/java/com/google/api/gax/batching/BatcherImpl.java#L126-L139),  So If the client is processing a large collection and one of the elements is null then 
 1. The client can either perform checks at [BatchingRequestBuilder](https://github.com/googleapis/gax-java/blob/f18673a1efb985243eee4a5cce01c7054ce18c23/gax/src/main/java/com/google/api/gax/batching/BatchingRequestBuilder.java#L44-L53) 
2. Pass it to callable and which would set NPE in the corresponding future. 

I assumed the latter could be a valid case and that's why I kept the getElement() Nullable. Please let me know if I should remove the `@Nullable`.

Reply:
Ok, that makes sense, but please mark the parameter element as Nullable in create as well

Reply:
Updated with the same.